cmd/apko/cli/publish.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"chainguard.dev/apko/pkg/build"
+	"chainguard.dev/apko/pkg/build/oci"
+	"chainguard.dev/apko/pkg/build/types"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func Publish() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "publish",
+		Short: "Build and publish an image",
+		Long: `Publish a built image from a YAML configuration file.
+
+It is assumed that you have used "docker login" to store credentials
+in a keychain.`,
```

Comment:
GGCR's `crane` has a `login` cobra command that is set up for downstream tools to link in.  We do this in cosign and ko IIRC.